blog/content/post/2020-02-13-rpi4_yocto.md

```diff
@@ -0,0 +1,272 @@
+
+```shell
+mkdir Yocto
+cd Yocto
+git clone -b zeus http://git.yoctoproject.org/git/poky
+```
+When cloning is finished we need to create a build directory and set up all the
+environmental variables, paths, etc. It can be obtained by typing
+
+```shell
+cd poky
+source oe-init-build-env
+```
+Yocto repository contains an extremally useful layer containing board support
+package called meta-raspberry pi that we need to add to our working directory
+
+```shell
+cd ~/Yocto
+git clone -b zeus git://git.yoctoproject.org/meta-raspberrypi
+```
+
+Before we build we need to make some changes in crucial configuration files
+Let's start from editing `bblayers.conf` located in `~/Yocto/poky/build`.
+This file tells bitbake about the location of various layers locations which
+are used to configure things like the target of a build and the location of
+output files. 
+
+Fist of all we need to add `meta-raspberrypi` layer we
+can do that by appending it to BBLAYERS. The resault should look like this
+
+```shell
+BBLAYERS ?= " \
+<projectPath>/Yocto/poky/meta \
+<projectPath>/Yocto/poky/meta-poky \
+<projectPath>/Yocto/poky/meta-yocto-bsp \
+<projectPath>/Yocto/meta-raspberrypi \
+" 
+```
+
+Then let's move to editing another file in the same directory. This time
+we dive into local.conf which is one of the most important files where
+many changes take place.
+
+First of all, we have to specify the machine on which we are working on.
+To do that just edit line containing `MACHINE ??=` to look like this:
+
+```shell
+MACHINE ??= "raspberrypi4-64.conf" 
+```
+in the same file, we can also add information about a number of threads 
+used to build an image. So append the file following lines (ideally) at
+the end of the local.conf:
+
+```shell
+BB_NUMBER_THREADS = "4"
+PARALLEL_MAKE = "-j 4" 
+``` 
+
+At this point, we are able to build an image but it would require some
+additional steps during flashing the image. By adding just a few lines we can
+make huge improvements:
+
+First of all, we can apply some features already provided by meta-raspberrypi.
+If we inspect `sdimage-raspberrypi.wks` file located in
+`Yocto/meta-raspberrypi/wic` we can see the following lines:
+
+```shell
+# short-description: Create Raspberry Pi SD card image
+# long-description: Creates a partitioned SD card image for use with
+# Raspberry Pi. Boot files are located in the first vfat partition.
+
+part /boot --source bootimg-partition --ondisk mmcblk0 --fstype=vfat --label boot --active --align 4096 --size 20
+part / --source rootfs --ondisk mmcblk0 --fstype=ext4 --label root --align 4096
+```
+
+If we add this to our project, an image will automatically create required
+partitions for boot and root on SD card after flashing. To provide that
+functionality everything we have to do is to append two lines to the local.conf
+
+```shell
+WKS_FILE = "sdimage-raspberrypi.wks"
+```
+
+This will tells Yocto to add `sdimage-raspberrypi.wks` to the project. And the
```

Comment:
to use given wks file when creating disk image, not to add wks file to the project